### PR TITLE
update masterview frame and pickerview frame where toolbar hidden; add t...

### PR DESCRIPTION
**Pull request by BruceZCQ**
update masterview frame and pickerview frame where toolbar hidden; add tap dismiss action  actionsheet's window. reflect didSucceedWithAction when tap action fired

### CODE REVIEW

**Comment by skywinder** [comment]
Hi. Thank you for pull request.

1. Can you explain it details, how to reproduce bug, that you fixed?

2. Please, remove from your pull-request all trailing spaces and unused changes - it's very hard to review code with a lot of redundant changes.

**Comment by BruceZCQ** [comment]
Sorry, this code 
```
// toolbar hidden remove the toolbar frame and update pickerview frame
    if (self.toolbar.hidden == YES) {
        masterView.frame = CGRectMake(0, 0, self.viewSize.width, 220);
        self.pickerView.frame = CGRectMake(0, 4, self.viewSize.width, 216);
    }
    
    [masterView addSubview:_pickerView];
    [self presentPickerForView:masterView];
    
    // add tap dismiss action
    self.actionSheet.window.userInteractionEnabled = YES;
    UITapGestureRecognizer *tapAction = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(onTapDismissAction)];
    [self.actionSheet.window addGestureRecognizer:tapAction];
```
added the function showActionSheetPicker bottom. update the frame when toolbar configed hidden. 

And add a function that when tap the window 
```
- (void)onTapDismissAction
{
    [self notifyTarget:self.target didSucceedWithAction:self.successAction origin:[self storedOrigin]];
    [self dismissPicker];
}
```

I joined the code in it, so the location of the other changes.

**Comment by skywinder** [comment]
@BruceZCQ thanks, this enhancement make sense.

Please, remove trailing spaces from lines, according our [contribution guideline](https://github.com/skywinder/ActionSheetPicker-3.0/blob/master/CONTRIBUTING.md) and [objective-c-style-guide](https://github.com/github/objective-c-style-guide):

> * Don’t leave trailing whitespace.
 *   Not even leading indentation on blank lines.

And I will ready to merge it.

**Comment by BruceZCQ** [comment]
@skywinder, removed trailing spaces from lines and modify the gitingore file

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+*pbxuser
+*perspectivev3
+*xcuserdata*
+# Junk files on Mac OS X
+*.DS_Store
+
+#build
+#include
+docs
+
```
Comment (skywinder) [inline .gitignore]: please, delete gitignore changes. In local gitignore you should story only project-specific ignore rules. Everything else should be placed in global gitignore.

File: CoreActionSheetPicker/CoreActionSheetPicker.xcodeproj/xcuserdata/BruceZCQ.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+			<true/>
+		</dict>
+		<key>1CA808781A25CE49004252BA</key>
+		<dict>
+			<key>primary</key>
+			<true/>
+		</dict>
+	</dict>
+</dict>
+</plist>
```
Comment (skywinder) [inline CoreActionSheetPicker/CoreActionSheetPicker.xcodeproj/xcuserdata/BruceZCQ.xcuserdatad/xcschemes/xcschememanagement.plist]: What for this changes?

File: Example-for-iOS-7-and-6/ActionSheetPicker-iOS6-7.xcodeproj/xcuserdata/BruceZCQ.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+	<key>SuppressBuildableAutocreation</key>
+	<dict>
+		<key>1D6058900D05DD3D006BFB54</key>
+		<dict>
+			<key>primary</key>
+			<true/>
+		</dict>
+	</dict>
+</dict>
+</plist>
```
Comment (skywinder) [inline Example-for-iOS-7-and-6/ActionSheetPicker-iOS6-7.xcodeproj/xcuserdata/BruceZCQ.xcuserdatad/xcschemes/xcschememanagement.plist]: And this?

File: Pickers/AbstractActionSheetPicker.m [hunk 4]
```diff
@@ -34,7 +34,7 @@ CG_INLINE BOOL isIPhone4()
 {
     struct utsname systemInfo;
     uname(&systemInfo);
-
+
```
Comment (skywinder) [inline Pickers/AbstractActionSheetPicker.m]: Here is a trailing spaces. And I can't merge it, unless this file got several trailing spaces.
please, avoid to put spaces in empty lines.

Comment (BruceZCQ) [inline Pickers/AbstractActionSheetPicker.m]: sorry, this code was not from me.

**Comment by BruceZCQ** [comment]
@skywinder  , I think I will pull request again for my code.

**Comment by BruceZCQ** [comment]
@skywinder , I created new pull request #140 . I am so sorry for my work.

**Comment by skywinder** [comment]
@BruceZCQ ok.

**Pull request closed without merging**